xen/arch/arm/coproc/coproc.c

```diff
@@ -330,9 +329,7 @@ struct coproc_device *coproc_alloc(struct platform_device *pdev,
     }
     coproc->dev = dev;
 
-    num_mmios = 0;
-    while ( (res = platform_get_resource(pdev, IORESOURCE_MEM, num_mmios)) )
-        num_mmios++;
+    num_mmios = dt_number_of_address(pdev);
```

Comment:
both dt_number_of_irq and dt_number_of_address return unsigned int. Why do you use int for them?